wallet/src/account/mod.rs

```diff
@@ -546,16 +546,29 @@ impl Account {
         &self,
         db_tx: &impl WalletStorageReadUnlocked,
         median_time: BlockTimestamp,
+        pool_id: PoolId,
     ) -> WalletResult<PoSGenerateBlockInputData> {
         let utxos = self.get_utxos(
             UtxoType::CreateStakePool | UtxoType::ProduceBlockFromStake,
             median_time,
             UtxoState::Confirmed.into(),
             WithLocked::Unlocked,
         );
-        // TODO: Select by pool_id if there is more than one UTXO
-        let (kernel_input_outpoint, (kernel_input_utxo, _token_id)) =
-            utxos.into_iter().next().ok_or(WalletError::NoUtxos)?;
+        let (kernel_input_outpoint, (kernel_input_utxo, _token_id)) = utxos
+            .into_iter()
+            .find(|(_kernel_input_outpoint, (kernel_input_utxo, _token_id))| {
+                pool_id
+                    == match kernel_input_utxo {
+                        TxOutput::CreateStakePool(pool_id, _) => *pool_id,
+                        TxOutput::ProduceBlockFromStake(_, pool_id) => *pool_id,
+                        TxOutput::Transfer(_, _)
+                        | TxOutput::LockThenTransfer(_, _, _)
+                        | TxOutput::Burn(_)
+                        | TxOutput::CreateDelegationId(_, _)
+                        | TxOutput::DelegateStaking(_, _) => panic!("Unexpected UTXO"),
+                    }
```

Comment:
Minor style suggestion: the `match` in the comparison expression here looks a bit strange. How about introducing a variable for it?